README.md

```diff
@@ -5,106 +5,143 @@ This is the server component of the AeroGear Metrics Service. It is a RESTful AP
+psql -U postgresql -d aerogear_mobile_metrics --host localhost
+Password for user postgresql: # postgres
+
+aerogear_mobile_metrics=> select * from mobileappmetrics;
 ```
 
-This will copy a binary from the default output location for `make build_linux` command, and build an image from it.
+## Builds and Testing
 
-## Release
+The `makefile` provided provides commands for building and testing the code. For now, only the most important commands are documented.
 
-Build and publish to github releases using `goreleaser`, see `.goreleaser.yml` for configuration.
+* `make setup` - Downloads the application dependencies.
 
-First make sure you have this tool installed: [Intalling Goreleaser](https://goreleaser.com/#introduction.installing_goreleaser).
+* `make build` - Compiles a binary for your current system. The binary is output at `./aerogear-app-metrics`.
 
-Then tag your release, replacing `x` with the appropriate version:
-```
-git tag -a v0.0.x -m "Release 0.0.x"
-```
+* `make build_linux` - Compiles a Linux compatible binary. This is mainly used for Docker builds. **Note:** `make build` should still be used if you are a Linux user.
 
-And make the release:
-```
-make release
-```
+* `make docker_build` - Builds a Binary from source and uses that binary to create a Docker image.
 
-### Releasing a built container
+* `make test` - Runs the unit tests (alias for `make test-unit`).
 
-This repository also contains a set of commands to push the built container images to the `aerogear` docker hub organization.
+* `make test-integration` - Runs unit tests and integration tests. This requires a running database server.
 
-These commands should preferrably run in a CD server, but are documented here for completeness and when required to be run locally
+* `make test-integration-cover` - Same as `make test-integration` but also generates a code coverage report. Used in the CI service.
 
-```
-# Add your dockerhub credentials to the environment variables
-DOCKERHUB_USERNAME={} DOCKERHUB_PASSWORD={} make docker_push_master
 
-# Manually add the tag information that would otherwise come from CI server metadata
-RELEASE_TAG=v0.0.1 make docker_push_release
-```
+## How to Release a New Version
+
+The release process for Aerogear maintainers is very simple. From the Github UI, simply create a release. **The release tag and title must be in the format `x.y.z`**. Formats such as `1.0` or `v1.0.0` are not valid.
+
+This will kick off an automated process in the CI service, where the following steps occur:
+
+* The release tag is checked out.
+* The code is built and the full test suite is run.
+* The [goreleaser](https://goreleaser.com/) tool generates a Changelog and binaries for Windows, MacOS and Linux. These are added to the release created in Github.
+* A new docker image is built and given the tags `latest` and `<git tag>` (where `<git tag>` is the `x.y.z` tag that was used).
+* The docker image is pushed to the Aerogear organization in Github.
```

Comment:
Not sure about capitalization for AeroGear, I see the 'g' in both lower and upper case in the website and other places, is there an official one?

For this repo we used `AeroGear` in CONTRIBUTING.md already.

Reply:
You're right, I think it's actually `Aerogear`. I've updated the Readme to only use this style.

Reply:
@matzew confirmed via IRC that it should be preferably 'AeroGear'